### ISSUE TITLE
issue when purchase non consumable

### ISSUE THREAD

**Issue by lesterjlester** [j3k0/cordova-plugin-purchase#1315]
Product (devgems100) does not exist or is not sucessfully initialized.

**Comment by wsamoht** [comment]
Did you [register](https://github.com/j3k0/cordova-plugin-purchase/blob/master/doc/api.md#registering-products) it first?

**Comment by lesterjlester** [comment]
> 

Yes, it should not work? If the status of in up purchase in appstore is ready for submit?

**Comment by lesterjlester** [comment]
We need wait to approved in up purchase first in appstore?

**Comment by wsamoht** [comment]
I was able to test with a dev build and in TestFlight with an internal tester without having it approved by the App Store first.

Otherwise, yes, Apple needs to review a build first with subscriptions.

From App Store Connect:
> Your first subscription must be submitted with a new app version. Create your subscription, then select it from the app’s In-App Purchase and Subscriptions section on the version page before submitting the version to App Review.
>
> Once your binary has been uploaded and your first subscription has been submitted for review, additional subscriptions can be submitted from the Subscriptions section.

**Comment by lesterjlester** [comment]
How. I can test in app purchase without approved?

**Comment by lesterjlester** [comment]
After being appoved by the appstore a in app purchase it  will work now on my app?

**Comment by wsamoht** [comment]
Like I said, if you load a dev build of your app onto an Apple device or do install it via TestFlight as an Internal Tester, you should be able test before having it approved on the app store. At least I was able to.

**Comment by j3k0** [comment]
This issue reports behavior from a pre-v13 version of the plugin. The relevant codepaths have been completely rebuilt since then. If you are seeing similar symptoms on the latest version, please open a new issue with reproduction details.
